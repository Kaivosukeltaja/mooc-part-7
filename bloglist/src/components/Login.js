import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { login, logout } from '../actions/users'

const initialState = {
  username: '',
  password: '',
}

class Login extends React.Component {
  constructor(props) {
    super(props)
    this.state = initialState
  }

  onSubmit = (event) => {
    event.preventDefault()
    this.props.login(this.state.username, this.state.password)
  } 

  onChange = (event) => {
    this.setState({ [event.target.name]: event.target.value })
  }  

  render() {
    return (
      <form onSubmit={this.onSubmit}>
        <h2>Log in to blog application</h2>
        <p>
          Username:{' '}
          <input
            name="username"
            value={this.state.username}
            onChange={this.onChange}
          />
        </p>
        <p>
          Password:{' '}
          <input
            type="password"
            name="password"
            value={this.state.password}
            onChange={this.onChange}
          />
        </p>
        <p>
          <input type="submit" value="Log in" />
        </p>
      </form>
    )
  }
}

Login.propTypes = {
  login: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
}

const mapDispatchToProps = {
  login,
  logout,
}

export default connect(null, mapDispatchToProps)(Login)
