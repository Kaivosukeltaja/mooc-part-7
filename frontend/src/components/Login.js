import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { login, logout } from '../actions/users'

const styles = {
  container: {
    height: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  heading: {
    borderBottom: '1px solid #eeeeee',
    margin: 0,
  },
  card: {
    backgroundColor: '#ffffff',
    boxShadow: '3px 3px 3px rgba(0,0,0,0.25)',
    padding: '1em 2em',
    borderRadius: '4px',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'stretch',
  },
  input: {
    margin: '0.5em 0',
    padding: '0.5em',
  },
  submit: {
    padding: '1em',
    margin: '0.5em 0',
  }
}

const initialState = {
  username: '',
  password: '',
}

class Login extends React.Component {
  constructor(props) {
    super(props)
    this.state = initialState
  }

  onSubmit = (event) => {
    event.preventDefault()
    this.props.login(this.state.username, this.state.password)
  } 

  onChange = (event) => {
    this.setState({ [event.target.name]: event.target.value })
  }  

  render() {
    return (
      <form onSubmit={this.onSubmit} style={styles.container}>
        <div style={styles.card}>
          <h2>Log in to blog application</h2>
          <input
            name="username"
            value={this.state.username}
            placeholder="Username"
            onChange={this.onChange}
            style={styles.input}
          />
          <input
            type="password"
            name="password"
            placeholder="Password"
            value={this.state.password}
            onChange={this.onChange}
            style={styles.input}
          />
          <input type="submit" value="Log in" style={styles.submit} />
        </div>
      </form>
    )
  }
}

Login.propTypes = {
  login: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
}

const mapDispatchToProps = {
  login,
  logout,
}

export default connect(null, mapDispatchToProps)(Login)
