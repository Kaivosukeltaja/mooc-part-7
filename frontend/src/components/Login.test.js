import React from 'react'
import renderer from 'react-test-renderer'
import { Login } from './Login'

describe('<Login />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Login login={() => null} logout={() => null} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
