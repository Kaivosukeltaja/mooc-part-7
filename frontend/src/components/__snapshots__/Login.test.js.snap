// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<form
  onSubmit={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "#ffffff",
        "borderRadius": "4px",
        "boxShadow": "3px 3px 3px rgba(0,0,0,0.25)",
        "display": "flex",
        "flexDirection": "column",
        "padding": "1em 2em",
      }
    }
  >
    <h2>
      Log in to blog application
    </h2>
    <input
      name="username"
      onChange={[Function]}
      placeholder="Username"
      style={
        Object {
          "margin": "0.5em 0",
          "padding": "0.5em",
        }
      }
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="Password"
      style={
        Object {
          "margin": "0.5em 0",
          "padding": "0.5em",
        }
      }
      type="password"
      value=""
    />
    <input
      style={
        Object {
          "margin": "0.5em 0",
          "padding": "1em",
        }
      }
      type="submit"
      value="Log in"
    />
  </div>
</form>
`;
